configure

```diff
@@ -69,22 +69,58 @@ Options:
 EOF
 }
 
-# This is just an example to generate a certfile for TLS distribution.
-# This is not an endorsement of specific expiration limits, key sizes, or algorithms.
+# Just an example to generate certificates for TLS distribution.
+# It's not an endorsement of specific expiration limits, key sizes, or algorithms.
 generate_tls_dev_cert() {
-    if [ ! -e "${rootdir}/dev/erlserver.pem" ]; then
-        openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout key.pem -out cert.pem
-        cat key.pem cert.pem > dev/erlserver.pem && rm key.pem cert.pem
-    fi
+    if [ "$ERLANG_VER" -lt 26 ]; then
+        if [ ! -e "${rootdir}/dev/erlserver.pem" ]; then
+            openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout key.pem -out cert.pem
+            cat key.pem cert.pem > dev/erlserver.pem && rm key.pem cert.pem
+        fi
 
-    if [ ! -e "${rootdir}/dev/couch_ssl_dist.conf" ]; then
-        cat > "${rootdir}/dev/couch_ssl_dist.conf" << EOF
-[{server,
-  [{certfile, "${rootdir}/dev/erlserver.pem"},
-   {secure_renegotiate, true}]},
- {client,
-  [{secure_renegotiate, true}]}].
+        if [ ! -e "${rootdir}/dev/couch_ssl_dist.conf" ]; then
+            cat >"${rootdir}/dev/couch_ssl_dist.conf" <<EOF
+[{server, [
+    {certfile, "${rootdir}/dev/erlserver.pem"},
+    {secure_renegotiate, true}
+  ]},
+  {client, [
+    {secure_renegotiate, true}
+  ]}
+].
 EOF
+        fi
+    else
+        if [ ! -e "${rootdir}/dev/erlserver.pem" ]; then
+            echo "Make sure to use different values for the Common Name (FQDN) in CA certificate and certificate."
+            read -p "Common Name (CA certificate): " fqdn_ca
+            read -p "Common Name (certificate): " fqdn
+
+            echo "Generate CA certificate:"
+            openssl genrsa 2048 >ca-key.pem
+            openssl req -new -x509 -nodes -days 3650 -key ca-key.pem -out ca-cert.pem -subj "/CN=$fqdn_ca"
+            echo "Generate certificate:"
+            openssl req -newkey rsa:2048 -nodes -days 3650 -keyout key.pem -out req.pem -subj "/CN=$fqdn"
+            openssl x509 -req -days 3650 -set_serial 01 -in req.pem -out cert.pem -CA ca-cert.pem -CAkey ca-key.pem
+            openssl verify -CAfile ca-cert.pem ca-cert.pem cert.pem
+            cat key.pem cert.pem >dev/erlserver.pem && mv ca-cert.pem dev/ && rm ca-key.pem key.pem cert.pem req.pem
+        fi
+
+        if [ ! -e "${rootdir}/dev/couch_ssl_dist.conf" ]; then
+            cat >"${rootdir}/dev/couch_ssl_dist.conf" <<EOF
+[{server, [
+    {certfile, "${rootdir}/dev/erlserver.pem"},
+    {secure_renegotiate, true}
+  ]},
+  {client, [
+    {secure_renegotiate, true},
+    {verify, verify_peer},
+    {cacertfile, "${rootdir}/dev/ca-cert.pem"},
+    {server_name_indication, "$fqdn"}
+  ]}
+].
```

Comment:
Wonder if it makes sense to use `{fail_if_no_peer_cert, true}` as well. Just guessing that users would go for a mTLS (mutual TLS) setup of some sort. Maybe something like this could work?

```
[{server, [
    {certfile, ".../erlserver.pem"},
    {cacertfile, ".../ca-cert.pem"},
    {verify, verify_peer},
    {secure_renegotiate, true},
    {fail_if_no_peer_cert, true}
  ]},
  {client, [
    {cacertfile, ".../ca-cert.pem"},
    {certfile, ".../erlserver.pem"},
    {verify, verify_peer},
    {secure_renegotiate, true},
    {server_name_indication, "localhost"}
  ]}
].
```

Reply:
I'll keep trying, thanks for the suggestion.